distributed/shuffle/shuffle_extension.py

```diff
@@ -578,3 +552,28 @@ def list_of_buffers_to_table(data: list[pa.Buffer], schema: pa.Schema) -> pa.Tab
     data = sr.read_all()
     bio.close()
     return data
+
+
+def deserialize_schema(data: bytes) -> pa.Schema:
+    """Deserialize an arrow schema
+
+    Examples
+    --------
+    >>> b = schema.serialize()  # doctest: +SKIP
+    >>> deserialize_schema(b)  # doctest: +SKIP
+
+    See also
+    --------
+    pa.Schema.serialize
+    """
+    import io
+
+    import pyarrow as pa
+
+    bio = io.BytesIO()
+    bio.write(data)
+    bio.seek(0)
+    sr = pa.RecordBatchStreamReader(bio)
+    table = sr.read_all()
+    bio.close()
+    return table.schema
```

Comment:
I was sad that Arrow doesn't have things like this builtin.  TODO: raise some JIRA tickets.